dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/State/TurnState.cs

```diff
@@ -1,121 +1,408 @@
+        /// Returns the TurnStateEntry associated with the specified currScope.
         /// </summary>
-        public const string TempStateKey = "tempState";
+        /// <param name="scope">The specified currScope.</param>
+        /// <returns>The state saved for the currScope.</returns>
+        public TurnStateEntry<Record> GetScope(string scope)
+        {
+            return _scopes[scope];
+        }
 
         /// <summary>
-        /// Stores all the conversation-related state entry.
+        /// Stores all the conversation-related state.
         /// </summary>
-        public TurnStateEntry<TConversationState>? ConversationStateEntry
+        public TConversationState? Conversation
         {
             get
             {
-                return Get<TurnStateEntry<TConversationState>>(ConversationStateKey);
+                TurnStateEntry<Record> scope = GetScope(CONVERSATION_SCOPE);
+                Verify.ParamNotNull(scope);
+                return scope.Value as TConversationState;
             }
             set
             {
                 Verify.ParamNotNull(value);
-
-                Set(ConversationStateKey, value!);
+                TurnStateEntry<Record> scope = GetScope(CONVERSATION_SCOPE);
+                Verify.ParamNotNull(scope);
+                scope.Replace(value);
             }
         }
 
         /// <summary>
-        /// Stores all the user related state entry.
+        /// Stores all the user related state.
         /// </summary>
-        public TurnStateEntry<TUserState>? UserStateEntry
+        public TUserState? User
         {
             get
             {
-                return Get<TurnStateEntry<TUserState>>(UserStateKey);
+                TurnStateEntry<Record> scope = GetScope(USER_SCOPE);
+                Verify.ParamNotNull(scope);
+                return scope.Value as TUserState;
             }
             set
             {
                 Verify.ParamNotNull(value);
-
-                Set(UserStateKey, value!);
+                TurnStateEntry<Record> scope = GetScope(USER_SCOPE);
+                Verify.ParamNotNull(scope);
+                scope.Replace(value);
             }
         }
 
         /// <summary>
-        /// Stores all the temporary state entry for the current turn.
+        /// Stores all the temporary state for the current turn.
         /// </summary>
-        public TurnStateEntry<TTempState>? TempStateEntry
+        public TTempState? Temp
         {
             get
             {
-                return Get<TurnStateEntry<TTempState>>(TempStateKey);
+                TurnStateEntry<Record> scope = GetScope(TEMP_SCOPE);
+                Verify.ParamNotNull(scope);
+                return scope.Value as TTempState;
             }
             set
             {
                 Verify.ParamNotNull(value);
+                TurnStateEntry<Record> scope = GetScope(TEMP_SCOPE);
+                Verify.ParamNotNull(scope);
```

Comment:
This method should just be called on parameters, not local variables. You should create a new method called `Verify.NotNull`.